transip/provider.go

```diff
@@ -0,0 +1,84 @@
+package transip
+
+import (
+	"context"
+	"time"
+
+	"github.com/libdns/libdns"
+	transipdomain "github.com/transip/gotransip/domain"
+)
+
+// Provider implements the libdns interfaces for Route53
+type Provider struct {
+	AccountName    string `json:"account_name"`
+	PrivateKeyPath string `json:"private_key_path"`
+	repository     transipdomain.Repository
+}
+
+// GetRecords lists all the records in the zone.
+func (p *Provider) GetRecords(ctx context.Context, zone string) ([]libdns.Record, error) {
+	records, err := p.getDNSEntries(ctx, zone)
+	if err != nil {
+		return nil, err
+	}
+
+	return records, nil
+}
+
+// AppendRecords adds records to the zone. It returns the records that were added.
+func (p *Provider) AppendRecords(ctx context.Context, zone string, records []libdns.Record) ([]libdns.Record, error) {
+	var appendedRecords []libdns.Record
+
+	for _, record := range records {
+		newRecord, err := p.addDNSEntry(ctx, zone, record)
+		if err != nil {
+			return nil, err
+		}
+		newRecord.TTL = time.Duration(newRecord.TTL) * time.Second
+		appendedRecords = append(appendedRecords, newRecord)
+	}
+
+	return appendedRecords, nil
+}
+
+// DeleteRecords deletes the records from the zone. If a record does not have an ID,
+// it will be looked up. It returns the records that were deleted.
+func (p *Provider) DeleteRecords(ctx context.Context, zone string, records []libdns.Record) ([]libdns.Record, error) {
```

Comment:
If TransIP doesn't assign IDs to records, then godocs about IDs don't make much sense IMO.

Reply:
No, this was a leftover from the Route53 implementation. Will remove